### PR TITLE
CI: Automatically generate releases when pushing tags

### PR DESCRIPTION
**Pull request by alessandrogario**

### CODE REVIEW

File: scripts/build.sh [hunk 1]
```diff
@@ -27,6 +27,7 @@ LLVM_VERSION=llvm-9
 OS_VERSION=
 ARCH_VERSION=
 BUILD_FLAGS=
+CXX_COMMON_VERSION="v0.1.1"
```
Comment (ekilmer) [inline scripts/build.sh:30]: I really should have done this at the beginning. Hopefully it wasn't too much headache 😬 

I _think_ `v0.1.2` should be all done now. Can we try that bump here?

File: .github/workflows/vcpkg_ci.yml [hunk 2]
```diff
@@ -127,4 +127,105 @@ jobs:
...
+        uses: actions/create-release@v1
+
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+        with:
+          tag_name: ${{ github.ref }}
+          release_name: Version ${{ github.ref }}
+          draft: true
+          prerelease: true
```
Comment (ekilmer) [inline .github/workflows/vcpkg_ci.yml:156]: Do these need to change?

Comment (alessandrogario) [reply .github/workflows/vcpkg_ci.yml:156]: This was done on purpose, so that we still have a chance to edit the release before it goes public

**Review by ekilmer — APPROVED** [review]
LGTM

**Pull request merged**